packages/material-ui/src/Menu/Menu.js

```diff
@@ -71,13 +101,44 @@ const Menu = React.forwardRef(function Menu(props, ref) {
     }
   };
 
+  const handleEntered = (element, isAppearing) => {
+    if (atLeastOneSubMenu) setEntering(false);
+
+    if (onEntered) {
+      onEntered(element, isAppearing);
+    }
+  };
+
+  const handleOnClose = (event) => {
+    event.preventDefault();
+    setOpenSubMenuIndex(null);
+    if (onClose) {
+      onClose(event, `${event.key.toLowerCase()}KeyDown`);
+    }
+  };
+
   const handleListKeyDown = (event) => {
-    if (event.key === 'Tab') {
-      event.preventDefault();
+    const { anchorEl } = other;
+    if (event.key === 'Tab' || event.key === 'Escape') {
+      handleOnClose(event);
+    }
 
-      if (onClose) {
-        onClose(event, 'tabKeyDown');
-      }
+    // This removes the imperatively added focusVisible class (from just below)
+    // when the user navigates away from the item and allows normal focus
+    // management to take back over.
+    anchorEl.onblur = () => anchorEl.classList.remove('Mui-focusVisible');
+
+    if (event.key === 'ArrowLeft' && isSubMenu) {
+      // This assigns the focusVisible class to the parent item when closing
+      // a sub menu using the left arrow keyboard navigation.
+      anchorEl.onfocus = async () => {
+        await anchorEl.classList.remove('MuiMenuItem-openSubMenuParent');
+        anchorEl.classList.add('Mui-focusVisible');
+      };
+      // Tell the parent Menu to close the sub Menu that you're in, but
+      // don't trigger the sub Menu onClose cascade.
+      if (!event.defaultPrevented) setParentOpenSubMenuIndex(null);
```

Comment:
I understand that we don't have a lint rule for this. Mostly because we rarely have to enforce it. But given the size of the PR, please wrapp all of these in brackets.

Reply:
Just to be clear, the issue is the inline `if`? So, make sure all `if` statements have a proper block around them?

Reply:
> So, make sure all if statements have a proper block around them?

:+1: 

```diff
- if (!event.defaultPrevented) setParentOpenSubMenuIndex(null);
+ if (!event.defaultPrevented) {
+   setParentOpenSubMenuIndex(null);
+ }

It just makes vertically scanning easier.

Reply:
Apologies, I'm realizing I missed several of your responses that were buried in collapsed parts of the conversation and I haven't been getting emails for all of these, it seems.

Cool, I'll address this and any other inline ifs I notice.